py/desispec/scripts/fit_cte_night.py

```diff
@@ -62,7 +64,27 @@ def main(args=None, comm=None):
         args.outfile = findfile("ctecorrnight", night=args.night, specprod_dir=args.specprod_dir)
 
     #- Create output directory if needed
-    os.makedirs(os.path.dirname(args.outfile), exist_ok=True)
+    if comm is None or comm.rank == 0:
+        os.makedirs(os.path.dirname(args.outfile), exist_ok=True)
+
+    #- Check what cameras are actually needed by science exposures
+    if args.expids is None:
+        etablefile = os.path.join(os.environ['DESI_SPECTRO_REDUX'],
+                          os.environ['SPECPROD'],
+                          'exposure_tables', str(args.night // 100),
+                          f'exposure_table_{args.night}.csv')
+        etable = load_table(etablefile, tabletype='exptable')
+        keep = etable['OBSTYPE'] == 'science'
+        sci_etable = etable[keep]
+        if len(sci_etable) == 0:
+            log.warning(f'No science exposures on {args.night}, but calculating CTE corrections anyway...')
+        else:
+            camword = parse_cameras(args.cameras)
```

Comment:
This works when `args.cameras` is a superset of CAMWORD, but will give incorrect results if CAMWORD is restricted but cameras is e.g. all cameras=a0123456789.

I think what you want is to find the GOODCAMWORD=CAMWORD-BADCAMWORD for each exposure, find the union of those, then take the intersection with `input_camword=parse_cameras(args.cameras)`. You can do the logic for GOODCAMWORD yourself, but there is a function that will take an erow and provide you with that. There is also a function for taking the intersection.

Useful functions for this:
`desispec.io.util.erow_to_goodcamword(erow, suppress_logging=True)`
`desispec.io.util.camword_intersection(camwords)`